### PR TITLE
Added support for PIVX

### PR DESCRIPTION
**Pull request by kr-mohak**
## Description

We have added support for [PIVX](https://pivx.org/), a bitcoin based blockchain. Since the transaction structure, signing process, address, etc. are same as bitcoin, we have not created new Address or Signer classes.

## Testing instructions

We have added all the common cpp, swift and kotlin tests.

## Types of changes
* New feature (non-breaking change which adds functionality)

## Checklist

- [ ] Prefix PR title with `[WIP]` if necessary.
- [x] Add tests to cover changes as needed.
- [ ] Update documentation as needed.

### CODE REVIEW

**Comment by hewigovens** [comment]
Could you please check this [bitcoin-forks-checklist](https://developer.trustwallet.com/wallet-core/newblockchain#bitcoin-forks-checklist)?, make sure you have all covered

**Comment by kr-mohak** [comment]
I've gone through the checklist and added all the missing tests. The PR is now ready for review. The android test might need to be re-run.

File: tests/interface/TWPivxTransactionTests.cpp [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    ); 
+}
+
+TEST(PivxTransaction, LockScripts) {
+ 
+    auto script = WRAP(TWBitcoinScript, TWBitcoinScriptBuildForAddress(STRING("DMkaY8mK98YgyDTYBZFLP3Rx1cMotAVA5Z").get()));
+    auto scriptData = WRAPD(TWBitcoinScriptData(script.get()));
+    assertHexEqual(scriptData, "76a914b637f1512b4517737009cfd3f7af3c9c692e13f088ac");
+ 
+    auto script2 = WRAP(TWBitcoinScript, TWBitcoinScriptBuildForAddress(STRING("D8dcUg3pjJy15tkC2MYKtnUsYJj83YyEjS").get()));
```
Comment (hewigovens) [inline tests/interface/TWPivxTransactionTests.cpp]: can you add a p2sh script test?

File: tests/interface/TWPivxTransactionTests.cpp [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    auto utxoKey0 = DATA("e2f7e8a1b0b0fa6934f0171a632ac3fc2c96901255ad27c2e97281e1359c76ef");
+    input.add_private_key(TWDataBytes(utxoKey0.get()), TWDataSize(utxoKey0.get()));
+
+
+    auto signer = TW::Bitcoin::TransactionSigner<TW::Bitcoin::Transaction>(std::move(input));
+    auto result = signer.sign();
+    auto signedTx = result.payload();
+
+    ASSERT_TRUE(result);
+    ASSERT_EQ(signer.plan.fee, 226);
```
Comment (hewigovens) [inline tests/interface/TWPivxTransactionTests.cpp]: For fees, do you prefer static fee or not? by default we will use `estimatefee` with in 2 blocks

**Comment by hewigovens** [comment]
Wait for https://github.com/TrustWallet/wallet-core/pull/521

File: tests/interface/TWPivxTransactionTests.cpp (hunk 2 above)
Comment (kr-mohak) [reply tests/interface/TWPivxTransactionTests.cpp]: We can go ahead with the defaults.

File: tests/interface/TWPivxTransactionTests.cpp (hunk 1 above)
Comment (kr-mohak) [reply tests/interface/TWPivxTransactionTests.cpp]: Done.

**Comment by hewigovens** [comment]
Hi, we add two new fields to `coins.json`:

1. `url`: project homepage url
2. `rpcNodeInfo`: rpc node software, for bitcion and forks, it's blockbook

Does blockbook support PIVX?

**Comment by hewigovens** [comment]
please also provide public blockbook node url, thanks!

File: coins.json [hunk 3]
```diff
@@ -804,5 +804,22 @@
...
+        "derivationPath": "m/44'/119'/0'/0/0",
+        "curve": "secp256k1",
+        "publicKeyType": "secp256k1",
+        "p2pkhPrefix": 30,
+        "p2shPrefix": 13,
+        "publicKeyHasher": "sha256ripemd",
+        "base58Hasher": "sha256d",
+        "xpub": "ppub",
+        "xprv": "pprv",
+        "explorer": "https://mainnet-explorer.pivx.org/tx/"
```
Comment (hewigovens) [inline coins.json]: add `url` and `rpcNodeInfo`

**Comment by kr-mohak** [comment]
> Does blockbook support PIVX?
Yes, PIVX is in [blockbook's  Port Registry](https://github.com/trezor/blockbook/blob/master/docs/ports.md). I don't think we need a separate url then, do we?

**Comment by kr-mohak** [comment]
Are there any issues still with the PR?

**Comment by hewigovens** [comment]
Sorry for late reply, we have an updated [requirements](https://developer.trustwallet.com/wallet-core/newblockchain#requirements) now, it looks like PIVX doesn’t rank top 100 in CMC.

**Comment by hewigovens** [comment]
Thanks for your efforts, let's cose it now

**Pull request closed without merging**